### ISSUE TITLE
Append version tag helper does not work for scoped CSS and blazor.webassembly.js

### ISSUE THREAD

**Issue by rogihee** [dotnet/aspnetcore#46798]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have a hosted WASM app in a razor page. The upgrade from dotnet 6.0 to dotnet 7.0 does not work if you have a cached `blazor.webassembly.js` file, the app gets stuck in the loading / boot sequence of blazor.

I thought I would use the `asp-append-version` tag helper to fix this in my Razor page where I host the app:

```
    <environment include="Development" exclude="Production">
        <script src="_framework/blazor.webassembly.js" autostart="false"></script>
    </environment>
    <environment include="Production" exclude="Development">
        <script src="_framework/blazor.webassembly.js" autostart="false" asp-append-version="true"></script>
    </environment>
```

After publishing this is rendered to:
```
<script src="_framework/blazor.webassembly.js" autostart="false"></script>
```

I expected something like:
```
<script src="_framework/blazor.webassembly.js?v=K1snUvACThylObaJIb1SuZcesNu0Kx348C9wL3nneWQ" autostart="false"></script>
```
**EDIT:** 
This also applies to the scoped CSS files generated by the SDK:

`<link href="MyApp.styles.css" rel="stylesheet" asp-append-version="true" />`

Renders to:
`<link href="MyApp.styles.css" rel="stylesheet" />`

Possibly related to #44216.

### Expected Behavior

I expected something like:
```
<script src="_framework/blazor.webassembly.js?v=K1snUvACThylObaJIb1SuZcesNu0Kx348C9wL3nneWQ" autostart="false"></script>
```

### Steps To Reproduce

1. Create a new hosted blazor webassembly project version 6.0
2. Add the version tag helper for the blazor.webassembly.js as above
3. Publish the site somewhere, say Azure Web App. 
4. Use Edge or Chrome browser to load the site with caching enabled.
5. Upgrade the project to 7.0 and publish again
6. Close the browser and restart at the same site
7. Observe that the app fails to load, stuck at loading with a console error. 

I have to try this with an isolated example, this is basically what happened to us in production. 

### .NET Version

9.0

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. It's not clear how the `blazor.webassembly.js` file got cached (the older version) on your clients' machines. Can you provide a repro step for making that happen, so that we can possibly suggest a workaround?

Also, maybe you can try to append `?1` to the filename and see if that works.

**Comment by rogihee** [comment]
@mkArtakMSFT the issue occurs when a client has a 6.0 runtime / blazor.webassembly.js cached in their browser and the website is upgraded to 7.0, then the loading fails in the boot sequence somewhere with a module error. Stuck at loading screen. I have modified the repro steps.

I have a workaround, I created this little version extension:
```
 private static string _version;

    public static string AppendVersion(this string url)
    {
        if (string.IsNullOrEmpty(_version))
        {
            _version = $"?v={Assembly.GetAssembly(typeof(BaseAsset))?.GetName().Version?.ToString() ?? ""}";
        }
        return $"{url}{_version}";
    }
```
This uses a verison set globally in the Directory.Build.props, so it is slightly less accurate than a file hash, but will do for my purposes. I use this now in the app.cshtml page:
```
<script src="@("_framework/blazor.webassembly.js".AppendVersion())"></script>
```

We'll see if this solves the problem. But this issue confirmed my **strong dislike of TagHelpers**: it can break without any warning or compile error, and a soft runtime error ("nothing") is a pretty horrible dev/deployment experience. Blazor United FTW!
